src/SourceLocator/Type/Composer/Factory/MakeLocatorForComposerJson.php

```diff
@@ -42,7 +42,7 @@ public function __invoke(string $installationPath, Locator $astLocator): SourceL
             throw MissingComposerJson::inProjectPath($installationPath);
         }
 
-        /** @var array{autoload: array{classmap: array<int, string>, files: array<int, string>, psr-4: array<string, array<int, string>>, psr-0: array<string, array<int, string>>}}|null $composer */
+        /** @var array{autoload: array{classmap: array<int, string>, files: array<int, string>, psr-4: array<string, array<int, string>|string>, psr-0: array<string, array<int, string>|string>}}|null $composer */
```

Comment:
Looking back at these, I really should've used `@psalm-type`, although I'm unsure if it even existed at the time...

Reply:
This fix is needed because Psalm reports it :)